internal/pkg/template/diff/write.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"io"
+)
+
+const (
+	prefixAdd = "+"
+	prefixDel = "-"
+	prefixMod = "~"
+)
+
+const indentInc = 4
+
+// Writer writes the string representation of a diff tree.
+type Writer struct {
+	tree   Tree
+	writer io.Writer
+}
+
+// Write uses the writer to write the string representation of the diff tree stemmed from the root.
+func (s *Writer) Write() error {
+	if s.tree.root == nil {
+		_, err := s.writer.Write([]byte("No changes.\n"))
```

Comment:
I wonder if we should instead just no write anything? And let the client decide how to display info back.

For example, here is what I'm imagining happens in the `cli` pkg:
```go
sb := new(strings.Builder)
tree, err := diff.Parse(deployedTemplate, pkgTemplate)
if err != nil { return err }
if err := tree.Write(sb); err != nil { return err }

out := sb.String()
if len(out) == "" {
   log.Info("There are no changes between the deployed stack and the local manifest for %q", o.name)
}
fmt.Println(out)
```